canvas_site_creator/views.py

```diff
@@ -27,6 +42,207 @@
 def lti_auth_error(request):
     raise PermissionDenied
 
+
+@login_required
+@lti_permission_required(canvas_api_accounts.ACCOUNT_PERMISSION_MANAGE_COURSES)
+@require_http_methods(['GET'])
+def index(request):
+    start_time = time.time()
+
+    sis_account_id = request.LTI['custom_canvas_account_sis_id']
+    ci_filters = {key: request.GET.get(key, '')
+                  for key in COURSE_INSTANCE_FILTERS}
+    (account_type, local_id) = canvas_api_accounts.parse_canvas_account_id(sis_account_id)
+    schools = []
+    terms = []
+    departments = []
+    course_groups = []
+    school = None
+
+    # Proceed only for valid account types
+    if account_type in canvas_api_accounts.ACCOUNT_TYPES:
+
+        # If user is in context Department or Course Group, display a
+        # message conveying restricted access
+        if (account_type == canvas_api_accounts.ACCOUNT_TYPE_DEPARTMENT or
+            account_type == canvas_api_accounts.ACCOUNT_TYPE_COURSE_GROUP):
+            return render(request, 'canvas_site_creator/restricted_access.html',
+                          status=403)
+
+        # We are in the context of a SIS type account, so limit options to that
+        # context
+        s = School.objects.get(pk=local_id)
+        school = {
+            'id': sis_account_id,
+            'name': s.title_short,
+        }
+
+        ci_filters[canvas_api_accounts.ACCOUNT_TYPE_SCHOOL] = school['id']
+        schools.append(school)
+
+    # if schools size is zero(includes improperly formatted account_type),
+    # display unauthorized message
+    if len(schools) == 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0` or `if len(val) &gt; 0`

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)